x/archivedb/codec.go

```diff
@@ -0,0 +1,29 @@
+package archivedb
+
+import (
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/codec/linearcodec"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+)
+
+const (
+	Version        = uint16(0)
+	maxMessageSize = 1 * units.MiB
+)
+
+var Codec codec.Manager
```

Comment:
I'd recommend against using the codec here. It's kind of overkill (because of reflection).